tools/linker/RemoveRejectedTypesStep.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Tuner;
+using MonoTouch.Tuner;
+using Xamarin.Tuner;
+
+namespace Xamarin.Linker {
+
+	public class RemoveRejectedTypesStep : RemoveCodeBase {
+
+		public RemoveRejectedTypesStep ()
+		{
+		}
+
+		public DerivedLinkContext Context {
+			get { return context as DerivedLinkContext; }
+		}
+
+		public override SubStepTargets Targets {
+			get { return SubStepTargets.Assembly; }
+		}
+
+		protected override string Name { get; } = " Removing Rejected Type";
+		protected override int ErrorCode { get; } = 2660;
+
+		public List<string> ReferencesToBeRemoved = new List<string> () {
+			// order is important for nested types
+			"UIKit.UIWebView/_UIWebViewDelegate",
+			"UIKit.UIWebView/UIWebViewAppearance",
+			"UIKit.IUIWebViewDelegate",
+			"UIKit.UIWebView",
+			"UIKit.UIWebViewDelegate",
+			"UIKit.UIWebViewDelegate_Extensions",
+			"UIKit.UIWebViewDelegateWrapper",
+			"UIKit.UIWebViewNavigationType",
+		};
+
+		protected override void Process (AssemblyDefinition assembly)
+		{
+			// this will give us a working `get_nse`
+			base.Process (assembly);
+
+			var updated = false;
+			if (Profile.IsProductAssembly (assembly)) {
+				updated = true;
+				foreach (var remove in ReferencesToBeRemoved) {
+					var type = assembly.MainModule.GetType (remove);
+					if (type == null)
+						continue;
+					type.Name = type.Name.Replace ("UI", "Deprecated");
```

Comment:
This only works for types named `UI*`. Maybe instead remove the first two characters? Or maybe make `ReferencesToBeRemoved` a hash table where the lookup value is the string to search for?

The whole purpose of this code falls down if this replacement fails, so it might make sense to detect if something wasn't replaced as expected (for instance check that pre- and post-replacement-strings are not identical) and show a warning in that case.

Reply:
It was not meant as a general purpose mechanism (since it's not clear if the next stone to hit us would fit the same pattern). However it means one less place to update if we supply to alternative name.

Reply:
hmm... List<T> guaranteed order (so inner types could be processed before their declared types). That's not true for a hash table or a (standard) dictionary.

Reply:
You could also create a list of tuples instead.

Reply:
yeah, that's what I'm doing. Time to move `mtouch` out of .net 4.6 :)